rdf-delta-server-local/src/main/java/org/seaborne/delta/server/local/patchstores/zk/PatchStoreZk.java

```diff
@@ -18,101 +18,82 @@
 import org.seaborne.delta.DeltaBadRequestException;
-import org.seaborne.delta.DeltaException;
 import org.seaborne.delta.server.local.LocalServerConfig;
 import org.seaborne.delta.server.local.PatchLog;
 import org.seaborne.delta.server.local.PatchStore;
 import org.seaborne.delta.server.local.PatchStoreProvider;
 import org.seaborne.delta.server.local.patchstores.PatchLogIndex;
 import org.seaborne.delta.server.local.patchstores.PatchStorage;
-import org.seaborne.delta.zk.Zk;
-import org.seaborne.delta.zk.Zk.ZkRunnable;
-import org.seaborne.delta.zk.Zk.ZkSupplier;
+import org.seaborne.delta.zk.UncheckedZkConnection;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class PatchStoreZk extends PatchStore {
     private final static Logger LOGZK = LoggerFactory.getLogger(PatchStoreZk.class);
-    // Use ServiceLoader.
-    public static final String PatchStoreZkName = "PatchStoreZk";
     // Schema!
     // https://curator.apache.org/curator-framework/schema.html
 
     // ---- Watching for logs
     private final Object storeLock = new Object();
 
-    private static AtomicInteger counter = new AtomicInteger(1);
-    private final CuratorFramework client;
-    private final InterProcessLock zkStoreLock;
+    private static final AtomicInteger counter = new AtomicInteger(1);
+    private final UncheckedZkConnection zk;
     private final String instance;
 
     // Our view of all the patch logs in the store.
-    private Map<String, PatchLog> patchLogs = new ConcurrentHashMap<>();
+    private final Map<String, PatchLog> patchLogs = new ConcurrentHashMap<>();
 
-    protected PatchStoreZk(CuratorFramework client, PatchStoreProvider psp) {
+    protected PatchStoreZk(UncheckedZkConnection client, PatchStoreProvider psp) {
         super(psp);
         this.instance = "zk-"+(counter.getAndIncrement());
-        this.client = client;
-        zkStoreLock = Zk.zkCreateLock(client, ZkConst.pStoreLock);
+        this.zk = client;
     }
 
-    public CuratorFramework getClient() { return client; }
+    public UncheckedZkConnection getClient() { return this.zk; }
     public String getInstance() { return instance; }
 
-    private void formatPatchStore(CuratorFramework client) throws Exception {
-        zkEnsure(client, ZkConst.pRoot);
-        zkEnsure(client, ZkConst.pStoreLock);
+    private void formatPatchStore() {
+        this.zk.ensurePathExists(ZkConst.pRoot);
+        this.zk.ensurePathExists(ZkConst.pStoreLock);
         // It does not matter if this holds up multiple log stores when they all run - it happens rarely.
-        Zk.zkLock(zkStoreLock, ZkConst.pStoreLock, ()->{
-            if ( ! zkExists(client, ZkConst.pLogs) )
-                zkCreate(client, ZkConst.pLogs);
-            if ( ! zkExists(client, ZkConst.pActiveLogs) )
-                zkCreate(client, ZkConst.pActiveLogs);
+        this.zk.runWithLock(ZkConst.pStoreLock, ()-> {
+            if ( ! this.zk.pathExists(ZkConst.pLogs) )
```

Comment:
These two lines should be one operation, equivalent to java.util.Map putIfAbsent.